flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -432,7 +441,76 @@ void EmitMLIRAction::ExecuteAction() {
     return;
   }
 
-  mlirModule.print(*os);
+  ci.mlirModule_.print(*os);
+}
+
+void EmitLLVMAction::ExecuteAction() {
+  CompilerInstance &ci = this->instance();
+
+  auto &ctx = ci.mlirCtx();
+  fir::support::loadDialects(ctx);
+  fir::support::registerLLVMTranslation(ctx);
+
+  // Set-up the pass manager
+  fir::setTargetTriple(ci.mlirModule_, "native");
+  auto &defKinds = ci.invocation().semanticsContext().defaultKinds();
+  fir::KindMapping kindMap(
+      &ci.mlirCtx(), llvm::ArrayRef<fir::KindTy>{fromDefaultKinds(defKinds)});
+  fir::setKindMapping(ci.mlirModule_, kindMap);
+  mlir::PassManager pm(&ci.mlirCtx(), mlir::OpPassManager::Nesting::Implicit);
+
+  pm.addPass(std::make_unique<Fortran::lower::VerifierPass>());
+  pm.enableVerifier(/*verifyPasses=*/true);
+  mlir::PassPipelineCLParser passPipeline("", "Compiler passes to run");
+
+  pm.addNestedPass<mlir::FuncOp>(fir::createArrayValueCopyPass());
+  pm.addNestedPass<mlir::FuncOp>(fir::createCharacterConversionPass());
+  pm.addPass(mlir::createCanonicalizerPass());
+  pm.addNestedPass<mlir::FuncOp>(fir::createCSEPass());
+  pm.addPass(mlir::createInlinerPass());
+  pm.addPass(mlir::createCSEPass());
+
+  pm.addNestedPass<mlir::FuncOp>(fir::createFirToCfgPass());
+  pm.addNestedPass<mlir::FuncOp>(fir::createControlFlowLoweringPass());
+  pm.addPass(mlir::createLowerToCFGPass());
+
+  pm.addPass(mlir::createCanonicalizerPass());
+  pm.addNestedPass<mlir::FuncOp>(fir::createCSEPass());
+  pm.addNestedPass<mlir::FuncOp>(fir::createAbstractResultOptPass());
+  fir::createFirTargetRewritePass(fir::TargetRewriteOptions{});
+  pm.addPass(fir::createFirTargetRewritePass(fir::TargetRewriteOptions{}));
+  pm.addPass(fir::createFIRToLLVMPass());
+
+  // If there is no pre-defined output stream, create an output file
+  std::unique_ptr<llvm::raw_ostream> os;
+  if (ci.IsOutputStreamNull()) {
+    // Lower from the LLVM dialect to LLVM IR
+    os = ci.CreateDefaultOutputFile(
+        /*Binary=*/true, /*InFile=*/GetCurrentFileOrBufferName(), "ll");
```

Comment:
I think `-emit-llvm` is more modal in `clang`, no? `-emit-llvm -c` emits a `.bc` file. `-emit-llvm -S` emits a `.ll` file. Etc. Is this emitting a bitcode file with a `.ll` extension? That may be surprising.